File: app_interface/lib/chatbot/models/content_model.dart
import 'package:cloud_firestore/cloud_firestore.dart';

class ContentModel {
  String id;
  String senderId;
  List<String> fileUrls;
  Timestamp timestamp;
  String output;
  Timestamp? ots;

  ContentModel({
    this.id = "",
    this.senderId = '',
    this.fileUrls = const [],
    required this.timestamp,
    this.output = '',
    this.ots = null,
  });

  factory ContentModel.fromFirestore(DocumentSnapshot data) {
    Map<String, dynamic> mapData = data.data() as Map<String, dynamic>;

    return ContentModel(
      id: data.id,
      senderId: mapData['senderId'] ?? 'mh_bhashini',
      fileUrls: mapData['fileUrls'].cast<String>() ?? [],
      timestamp: mapData['timestamp'] ?? Timestamp.now(),
      output: mapData['output'] ?? '',
      ots: mapData['ots'] ?? null,
    );
  }

  Map<String, dynamic> getMap() {
    return {
      'id': id,
      'senderId': senderId,
      'fileUrls': fileUrls,
      'timestamp': timestamp,
      'output': output,
      'ots': ots,
    };
  }
}

File: app_backend/models/content_model.py
class ContentModel():
    def __init__(self, id, senderId,fileUrls,timestamp,output,ots):
        self.id = id
        self.senderId = senderId
        self.fileUrls = fileUrls
        self.timestamp = timestamp
        self.output=output
        self.ots=ots
    
    def from_dict(self,data):
        return ContentModel(
            id=data.get('id'),
            senderId=data.get('senderId'),
            fileUrls=data.get('fileUrls'),
            timestamp=data.get('timestamp'),
            output=data.get('output'),
            ots=data.get('ots'),
            )
    
